packages/eslint-config-react-app/index.js

```diff
@@ -174,8 +160,8 @@ module.exports = {
     'no-template-curly-in-string': 'warn',
     'no-this-before-super': 'warn',
     'no-throw-literal': 'warn',
-    'no-undef': 'error',
-    'no-restricted-globals': ['error'].concat(restrictedGlobals),
+    'no-undef': 'warn',
+    'no-restricted-globals': ['warn'].concat(restrictedGlobals),
```

Comment:
This, too, should stay an error. If you write code like

```js
handleChange() {
  this.setState({text: event.target.value});
}
```

your build should break. Otherwise you'll spend hours chasing this accidental global usage.

It's just not worth it to allow the build.